From fa68ef6100c158ddd2419355b3bddc71b564fe7f Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Thu, 2 Oct 2025 19:49:05 +0300
Subject: [PATCH 343/553] QFactoryLoader: fix clang -Wshorten-64-to-32 warnings

Drive-by, de-duplicate a lengthy line of code.

Pick-to: 6.8 6.5
Change-Id: I29d2846ba8cfcd333bee82653e98761bb19ccc2f
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 1084ef0094e152aa4a89e1f5519ced8d865c68d1)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/plugin/qfactoryloader.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/corelib/plugin/qfactoryloader.cpp b/src/corelib/plugin/qfactoryloader.cpp
index 48420219eaa..57a7a752424 100644
--- a/src/corelib/plugin/qfactoryloader.cpp
+++ b/src/corelib/plugin/qfactoryloader.cpp
@@ -275,6 +275,10 @@ inline void QFactoryLoader::Private::updateSinglePath(const QString &path)
 #endif
                 QDirListing::IteratorFlag::FilesOnly | QDirListing::IteratorFlag::ResolveSymlinks);
 
+    auto versionFromLib = [](const QLibraryPrivate *lib) {
+        return lib->metaData.value(QtPluginMetaDataKeys::QtVersion).toInteger();
+    };
+
     for (const auto &dirEntry : plugins) {
         const QString &fileName = dirEntry.fileName();
 #if defined(Q_PROCESSOR_X86)
@@ -313,7 +317,7 @@ inline void QFactoryLoader::Private::updateSinglePath(const QString &path)
             continue;
 
         static constexpr qint64 QtVersionNoPatch = QT_VERSION_CHECK(QT_VERSION_MAJOR, QT_VERSION_MINOR, 0);
-        int thisVersion = library->metaData.value(QtPluginMetaDataKeys::QtVersion).toInteger();
+        qint64 thisVersion = versionFromLib(library.get());
         if (iid.startsWith(QStringLiteral("org.qt-project.Qt.QPA"))) {
             // QPA plugins must match Qt Major.Minor
             if (thisVersion != QtVersionNoPatch) {
@@ -337,7 +341,7 @@ inline void QFactoryLoader::Private::updateSinglePath(const QString &path)
                 // If the existing library was built with a future Qt version,
                 // whereas the one we're considering has a Qt version that fits
                 // better, we prioritize the better match.
-                int existingVersion = existingLibrary->metaData.value(QtPluginMetaDataKeys::QtVersion).toInteger();
+                qint64 existingVersion = versionFromLib(existingLibrary);
                 if (existingVersion == QtVersionNoPatch)
                     continue; // Prefer exact Qt version match
                 if (existingVersion < QtVersionNoPatch && thisVersion > QtVersionNoPatch)
-- 
2.51.2

